ilastik/shell/gui/ilastikShell.py

```diff
@@ -1010,10 +1009,12 @@ def show(self):
         totalSplitterHeight = sum(self.sideSplitter.sizes())
         self.sideSplitter.setSizes([totalSplitterHeight / 2, totalSplitterHeight / 2])
 
+    @threadRouted
     def updateShellProjectDisplay(self):
         """
         Update the title bar and allowable shell actions based on the state of the currently loaded project.
         """
+        assert threading.current_thread().name == "MainThread"
```

Comment:
This is probably redundant because `ThreadRouter/threadRouted` always emits signals that go to the GUI thread.